components/team_sidebar/team_sidebar_controller.jsx

```diff
@@ -53,13 +54,76 @@ export default class TeamSidebar extends React.PureComponent {
         actions: PropTypes.shape({
             getTeams: PropTypes.func.isRequired,
             switchTeam: PropTypes.func.isRequired,
+            updateTeamsOrderForUser: PropTypes.func.isRequired,
         }).isRequired,
+        userTeamsOrderPreference: PropTypes.string,
+    }
+
+    constructor(props) {
+        super(props);
+        this.state = {
+            teamsOrder: filterAndSortTeamsByDisplayName(
+                this.props.myTeams,
+                this.props.locale,
+                this.props.userTeamsOrderPreference,
+            ),
+        };
+    }
+
+    static getDerivedStateFromProps(nextProps, prevState) {
+        const newTeamsOrder = filterAndSortTeamsByDisplayName(
+            nextProps.myTeams,
+            nextProps.locale,
+            nextProps.userTeamsOrderPreference
+        );
+        if (newTeamsOrder !== prevState.teamsOrder) {
```

Comment:
`filterAndSortTeamsByDisplayName` returns a new array each time it will be called, so `newTeamsOrder !== prevState.teamsOrder` will always be true.

Instead of storing `teamsOrder` in state, I think we could just call `filterAndSortTeamsByDisplayName` whenever we need it (in `render` and `onDragEnd`). This component will most likely only be re-rendered when one of those props changes, so it saves us from calling the function when those props haven't changed.

If performance ever becomes an issue with that, we could also have the component memoize the function, but that hopefully shouldn't be necessary.